gradle-guide-error-prone/src/test/java/com/palantir/gradle/guide/errorprone/CallGraphBugCheckerTest.java

```diff
@@ -0,0 +1,488 @@
+        assertTrue(containsMethodNamed(callees, "fourth"), "should call fourth (chained)");
+    }
+
+    @Test
+    void testLongMethodChain() throws Exception {
+        String javaCode =
+                """
+            class TestClass {
+                public void longChain() {
+                    getValue()
+                        .toString()
+                        .trim()
+                        .toLowerCase()
+                        .substring(0, 5)
+                        .replace("a", "b")
+                        .concat("suffix");
+                }
+
+                public Object getValue() { return "example"; }
+            }
+            """;
+
+        VisitorState state = parseJavaCode(javaCode);
+        MethodCallGraph callGraph = checker.new MethodCallGraph(state);
+        ValueGraph<MethodSymbol, Set<MethodInvocationTree>> graph = peekInternalGraph(callGraph);
+
+        MethodSymbol longChain = findMethodByName(graph.nodes(), "longChain");
+        assertNotNull(longChain, "longChain method should be in graph");
+
+        Set<MethodSymbol> callees = graph.successors(longChain);
+        assertTrue(containsMethodNamed(callees, "getValue"), "should call getValue");
+        assertTrue(containsMethodNamed(callees, "toString"), "should call toString (chained)");
+        assertTrue(containsMethodNamed(callees, "trim"), "should call trim (chained)");
+        assertTrue(containsMethodNamed(callees, "toLowerCase"), "should call toLowerCase (chained)");
+        assertTrue(containsMethodNamed(callees, "substring"), "should call substring (chained)");
+        assertTrue(containsMethodNamed(callees, "replace"), "should call replace (chained)");
+        assertTrue(containsMethodNamed(callees, "concat"), "should call concat (chained)");
+
+        // Verify we have the expected number of direct calls
+        long methodCallCount = callees.stream()
+                .filter(method -> !method.getSimpleName().toString().equals("longChain"))
+                .count();
+        assertEquals(7, methodCallCount, "should have exactly 7 direct calls");
+    }
+
+    @AutoService(BugChecker.class)
+    @BugPattern(summary = "", severity = SeverityLevel.SUGGESTION)
+    private static final class TestableCallGraphBugChecker extends CallGraphBugChecker {
+        @Override
+        public MoreInfoLink moreInfoLink() {
+            return null;
+        }
+    }
+
+    public static MutableValueGraph<MethodSymbol, Set<MethodInvocationTree>> peekInternalGraph(MethodCallGraph graph)
+            throws NoSuchFieldException, IllegalAccessException {
+        Class<?> methodCallGraphClass = graph.getClass();
+        Field callGraphField = methodCallGraphClass.getDeclaredField("callGraph");
+        callGraphField.setAccessible(true);
+        return (MutableValueGraph<MethodSymbol, Set<MethodInvocationTree>>) callGraphField.get(graph);
+    }
+
+    private VisitorState parseJavaCode(String javaCode) throws IOException {
```

Comment:
As discussed in person, let's try to not maintain all this errorprone initialization code. Maybe let's try using the `// BUG:` mechanism to validate our call graph. 

@felixdesouza and I will discuss it tomorrow.